src/arithmetic/arithmetic_expression.c

```diff
@@ -499,36 +492,45 @@ SIValue AR_TOLOWER(SIValue *argv, int argc) {
     assert(argc == 1);
 
     if(SIValue_IsNull(argv[0])) return SI_NullVal();
-    char *original = argv[0].stringval.str;
-    size_t lower_len = argv[0].stringval.len + 1;
-    char *lower = calloc(lower_len, sizeof(char));
+    char *original = argv[0].stringval;
+    size_t lower_len = strlen(argv[0].stringval) + 1;
+    char lower[lower_len];
     _toLower(original, lower, &lower_len);
-    return SI_StringValC(lower);
+    return SI_StringVal(lower);
+}
+
+void _toUpper(const char *str, char *upper, size_t *upper_len) {
+    size_t str_len = strlen(str);
+    /* Avoid overflow. */
+    assert(*upper_len > str_len);
+
+    /* Update upper len*/
+    *upper_len = str_len;
+
+    int i = 0;
+    for(; i < str_len; i++) upper[i] = toupper(str[i]);
+    upper[i] = 0;
 }
 
 SIValue AR_TOUPPER(SIValue *argv, int argc) {
     assert(argc == 1);
 
     if(SIValue_IsNull(argv[0])) return SI_NullVal();
-    char *original = argv[0].stringval.str;
-    size_t original_len = argv[0].stringval.len;
-    char *upper = calloc(original_len+1, sizeof(char));
-    
-    for(int i = 0; i < original_len; i++) {
-        upper[i] = toupper(original[i]);
-    }
-
-    return SI_StringValC(upper);
+    char *original = argv[0].stringval;
+    size_t upper_len = strlen(argv[0].stringval) + 1;
+    char upper[upper_len];
```

Comment:
upper[upper_len] might not contain 0 byte (null termination).
Initialise by
upper[upper_len] = {0};